content-src/components/TopSites/TopSites.js

```diff
@@ -26,22 +26,42 @@ const TopSitesItem = React.createClass({
     const site = this.props;
     const index = site.index;
     const isActive = this.state.showContextMenu && this.state.activeTile === index;
-    const screenshot = site.screenshot;
+
+    const topSitesExperimentIsOn = this.props.showNewStyle;
+    const screenshot = topSitesExperimentIsOn && site.screenshot;
 
     // The top-corner class puts the site icon in the top corner, overlayed over the screenshot.
     const siteIconClasses = classNames("tile-img-container", {"top-corner": screenshot});
 
+    let faviconSize = 32;
+    if (topSitesExperimentIsOn && !screenshot) {
+      if (!site.favicon_width) {
+        // default to 64 if not specified (*cough* tippy top)
```

Comment:
Not sure where this ended up in the IRC discussion.  Maybe this wants to change?